jerry-core/api/jerry.c

```diff
@@ -3088,6 +3089,75 @@ jerry_typedarray_find_by_type (jerry_typedarray_type_t type_name, /**< type of t
 
 #endif /* !CONFIG_DISABLE_ES2015_TYPEDARRAY_BUILTIN */
 
+/**
+ * Creates a RegExp object with the given pattern and flags.
+ *
+ * @return value of the Constructed RegExp object.
+ */
+jerry_value_t
+jerry_create_regexp (const jerry_char_t *pattern, /**< RegExp pattern */
+                    jerry_length_t pattern_length, /**< Length of the pattern */
+                    jerry_regexp_flags_t flags) /**< Optional RegExp flags */
+{
+  jerry_assert_api_available ();
+
+#ifndef CONFIG_DISABLE_REGEXP_BUILTIN
+  if (!lit_is_valid_utf8_string ((lit_utf8_byte_t *) pattern, (lit_utf8_size_t) pattern_length))
+  {
+    return jerry_throw (ecma_raise_common_error ("Input must be a valid utf8 string"));
+  }
+
+  ecma_string_t *ecma_pattern = ecma_new_ecma_string_from_utf8 ((lit_utf8_byte_t *) pattern,
+                                                                      (lit_utf8_size_t) pattern_length);
+
+  ecma_value_t str_val = ecma_make_string_value (ecma_pattern);
+
+  jerry_value_t ret_val;
+
+  if (flags > 0)
+  {
+    ecma_value_t flag_val = ECMA_VALUE_UNDEFINED;
+    jerry_char_t flag_str[3];
+    unsigned char *str_ptr = flag_str;
+
+    if (flags & JERRY_RE_FLAG_GLOBAL)
+    {
+      *str_ptr++ = 'g';
+    }
+
+    if (flags & JERRY_RE_FLAG_MULTILINE)
+    {
+      *str_ptr++ = 'm';
+    }
+
+    if (flags & JERRY_RE_FLAG_IGNORE_CASE)
+    {
+      *str_ptr++ = 'i';
+    }
+
+    flag_val = jerry_create_string_sz (flag_str, (jerry_size_t) (str_ptr - flag_str));
+
+    ecma_value_t args[2] = {str_val, flag_val};
+    ret_val = ecma_builtin_regexp_dispatch_construct (args, 2);
```

Comment:
I don't see why has this been marked as resolved. The code hasn't been rewritten to use `ecma_op_create_regexp_object` instead of `ecma_builtin_regexp_dispatch_construct`. This comment still applies, even if my other `ecma_op_create_regexp_object`-related comment is disputed.

Reply:
It has been marked to resolve since I've already answered the same part below, in the other comment.